File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> nums)
{
    
    int arr[200001]={0,};
    
    for(int i = 0 ; i < nums.size();i++){
        arr[nums[i]]++;
    }
    int kinds = 0;

    for(int i = 1 ; i <200001;i++){
        if(arr[i]>0){
            kinds++;
        }        
    }

    if(kinds > nums.size()/2){
        return nums.size()/2;
    }else{
        return kinds;
    }
    
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
class Solution {
    public int solution(int[] nums) {
        
        int [] arr = new int [200001];
        
    
        for(int i = 0 ; i < nums.length; i++){
            arr[nums[i]]=1;
        }
        
        int ans = 0;
        for(int i = 0 ; i < 200001; i++){
            if(arr[i]==1){
                ans++;
            }
        }
        
        int answer = 0;

        if(ans <= (nums.length/2)){
            answer = ans;
        }else if( ans > (nums.length/2)){
            answer = (nums.length / 2);
        }
        
        
        
        return answer;
        
    }
}